---
# TODO: make plots transparent
title: ""
subtitle: Reconstruction of draft genomes and full length rRNA genes from plant root
  and nodule metagenomes
author: "Mattias de Hollander"
date: "April 20, 2016"
output:
  ioslides_presentation:
    css: style.css
    # Change to svg for highres images
    dev: png
    fig_height: 5
    fig_width: 10
    incremental: no
    logo: images/NIOOlogov9.svg
    self_contained: yes
    widescreen: yes
---


# Introduction

## Plant root and nodule metagenomes

<div class="centered">
<img src="images/backtotheroots.png" width = 700/>  
</div>

## Endophytic metagenome (NL)
<!--
- conducive / suppressive
- CUR /EST (salt / non-salt)
- shannon diversity: nodules, endo, rhizo, bulk
- 16S extracted metagenomes OR gene classification
- difference 16S primers / 16S shotgun
-->
<div class="centered">
<img src="images/suppresive_soil.png" width = 800/>  
</div>

<reference>Weller, D.M. et.al (2002). Microbial Populations Responsible for Specific Soil Suppressiveness to Plant Pathogens. Annual Review of Phytopathology 40, 309–348.</reference>


## Nodule metagenome (Brazil)

<div class="centered">
<img src="images/marcio/nodule_sampling.png" width = 880/>  
</div>

<slidethanks>Márcio Leite</slidethanks>

<div class="notes">
We selected the leguminous plant specie Mimosa pigra L. sub-family Mimosoideae, tribe Eumimoseae and sampled in two sampling regions: one with the presence of saltwater intrusion (Curral da Igreja located 4.2 km from the mangrove region) and another one located far from the saltwater intrusion (Estirão Grande placed at 57.2 km further river upstream). 
</div>


## Sequencing details

Feature              Nodule           Endo   
------------- -------------  -------------
Platform              MiSeq  MiSeq + HiSeq
Read length             300            300
Samples                  18             10
Raw                8.32 Gbp      191.6 Gbp    
QC                 4.16 Gbp       47.2 Gbp
Gbp sample             0.23           4.72

## Diversity plots
<!--
- endo vs rhizo
- diversity nodules
-->

```{r,echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
library(phyloseq)
load('data/1511KMI-0007.RData')
glom_filt = filter_taxa(glom_class, function(x) sum(x) > 500, TRUE)
plot_bar(glom_filt, "Sample", "Abundance", "Class")
```

# Assembly

## Analysis pipeline

<div class="centered">
<img src="images/workflow_overview.svg" width = 850/>  
</div>

<reference>https://gitlab.bioinf.nioo.knaw.nl/shotgun-metagenomics/nodule-metagenome</reference>

## Assembly results
<!--- 
samtools flagstat
mmstats
bandage
--->

Feature                   Nodule           Endo   
------------------ -------------  -------------
assembler                megahit         SPAdes
% mapped                   99.65
properly paired            78.16
N50                       10,631          1,652
Average cov                  1.6
\# essential genes         5,769
Nodes                    133,461                    
Edges                    152,761                       
                                

# Annotation

## pipeline

## pictures victorc

```{r,echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
load("data/endo.RData")
endo_plot            <- plot_ordination(endo_metg, endo_total.ord, type = "sites",
                                        color = "Treatment",
                                        shape = "Treatment", 
                                        title = NULL, label = NULL)
endo_plot            = endo_plot  + geom_point(aes(shape=Treatment),   
                                               size = 8) + geom_text(aes(label = Treatment), size=0) +
        theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        #panel.background = element_blank(), axis.line = element_line(size = 1, colour = "black"),
        rect = element_rect(fill = "white",  colour = "black",
                            size = 1, linetype = 2),
        axis.ticks = element_line(colour = "black"),
        axis.title.x = element_text(face="bold", colour="black", size=30),
        axis.title.y = element_text(face="bold", colour="black", size=30),
        axis.text.x  = element_text(face="bold", colour = "black", angle=360, vjust=0.5, size=20),
        axis.text.y  = element_text(face="bold", colour = "black", angle=360, vjust=0.5, size=20),
        legend.background = element_rect(colour = "white"), 
        legend.key = element_rect(fill = "white", colour = "white"), 
        legend.key.height = NULL,
        legend.key.width = NULL, 
        legend.text = element_text(size = rel(1), colour = "white"), 
        legend.text.align = NULL, 
        legend.title = element_text(size = rel(0.5), face = "bold", colour = "white" ,hjust=0), 
        legend.title.align = NULL, 
        legend.position = c(0.1,0.9), 
        legend.direction = NULL, 
        legend.justification = "center", 
        legend.box = NULL,
        panel.background = element_rect(fill = "white", colour = "black", linetype = 1, size = 1.5),
        panel.border = element_blank(), 
        panel.grid.major = element_line(colour = "black"), 
        panel.grid.minor = element_line(colour = "black", size = 0.25))


plot(endo_plot)
```

<slidethanks>Victor Carrión</slidethanks>

# Binning

```{r,echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
library(mmgenome)
library(ggplot2)
library(RColorBrewer)
# Load gtools for the mixedsort function
library(gtools)

load('data/mmgenome.RData')
```

## 

<div class="centered">
<img src="images/workflow_binning.png" width = 600/>  
</div>

<reference>Sangwan, N., Xia, F., and Gilbert, J.A. (2016). Recovering complete and draft population genomes from metagenome datasets. Microbiome 4, 8.</reference>


## {.notitle}
<style type = "text/css">
.notitle {color: red;}
</style>

<div class="centered">
<img src="images/albertsen.jpg" width = 700/>  
</div>

<slidethanks>Mad Albertsen</slidethanks>

## 

<div class="centered">
<img src="images/mmgenome.jpg" width = 700/>  
</div>

## GC vs coverage
```{r gccov,echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
p <- mmplot(data = d, x = "gc", y = "MPR1.1n.EST", log.x = F, log.y = T, color = "essential", minlength = 200)
p
```

## 16S
```{r 16S,echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
ssu = read.table("data/contigs.ssu.txt", header = F, sep = "\t")$V1
highlight <- as.character(ssu)
sdata <- subset(d$scaffolds, scaffold %in% highlight)
p + geom_point(data = sdata, color = "darkred", size = 3, shape = 8)
```

## Colored bins
```{r bins, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
# Sort the bins by name
# Note that the dot in the bin name of metabat needs to be removed from the input
d$scaffolds$bin_sorted <- factor(d$scaffolds$bin, levels = mixedsort(levels(d$scaffolds$bin)))
colourCount = length(unique(d$scaffolds$bin_sorted))
# Set1 and Dark2 work
getPalette = colorRampPalette(brewer.pal(8, "Dark2"))

p <- mmplot(data = d, x = "gc", y = "MPR1.1n.EST", log.x = F, log.y = T, color = "bin_sorted", minlength = 5000)
p <- p + scale_colour_manual(values=getPalette(colourCount))
# Position the legend
p <- p + theme(legend.position="right", legend.box = "horizontal") + guides(colour = guide_legend(ncol = 2, override.aes = list(alpha = 1, size = 5, shape = 19)))
#p <- p + ylab("Coverage") + xlab("GC") + theme_gray(base_size = 24)
p
#ggplotly()
```

## Tetra nucleotide
```{r, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
p <- mmplot(data = d, x = "PC1", y = "PC2", log.x = F, log.y = F, color= "none", minlength = 3000)
p
```

## Tetra colored
```{r, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
p <- mmplot(data = d, x = "PC1", y = "PC2", log.x = F, log.y = F, color= "bin_sorted", minlength = 3000)
p <- p + scale_colour_manual(values=getPalette(colourCount))
# Position the legend
p <- p + theme(legend.position="right", legend.box = "horizontal") + guides(colour = guide_legend(ncol = 2, override.aes = list(alpha = 1, size = 5, shape = 19)))
p
```

## Completeness/contamination Checkm

Bin Id  Marker Lineage       Completeness  Contamination     
------  -------------------  ------------- -------------
bin.3   f__Paenibacillaceae          99.85	        0.02
bin.8   o__Actinomycetales           99.81	           0
bin.7   f__Moraxellaceae             99.51	        0.82
bin.9   f__Moraxellaceae             99.45	        0.14
bin.4   g__Burkholderia              98.46	        0.57
bin.10  o__Bacillales                97.80	        1.34


## Conclusion

- increasing diversity nodule, endo, rhizo
- extract full length 16S
- difference 16S/ metagenome
- automated/manual binning
- 16S in bins

## Acknowledgements
<!--- two columns? --->
- Vittorio Tracanna
- Noriko Cassman
- Victor Carrion
- Marcio Leite
- Eiko Kuramae
- Jos Raaijmakers

<div class="right">
<img src="images/rmarkdown.png" width=400>
</div>

# 

## Average Genome Size

```{r ags, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
ags = read.table('data/ags.txt',header=TRUE)

p <- ggplot(data=ags, aes(x=1:dim(ags)[1],y=average_genome_size.)) +
                                                                                                                                                                                    geom_bar(stat="identity")
p
#ggplotly()
```

## Gene length

```{r genelength, echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE}
d = read.table('data/length_abundsum.txt')
plot(d)
```

## Rhizo vs Endo

- plot diversity

## Image test

![](images/environments.png){#id .class width=30px height=2px}

## Image test2
<!-- This works -->
<img src="images/environments.png" width = 400/>  

## Slide with centered image and citation

<div class="centered">
<img src="img/lindquist-wager-fig-4.7.jpg" height=500px>

[@lindquist_wager_2015]
</div>